internal/service/invoice.go

```diff
@@ -783,3 +787,133 @@ func (s *invoiceService) performPaymentAttemptActions(ctx context.Context, inv *
 
 	return nil
 }
+
+// GetInvoicePDF implements InvoiceService.
+func (s *invoiceService) GetInvoicePDF(ctx context.Context, id string) ([]byte, error) {
+	// get invoice by id
+	inv, err := s.InvoiceRepo.Get(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+
+	// fetch customer info
+	customer, err := s.CustomerRepo.Get(ctx, inv.CustomerID)
+	if err != nil {
+		return nil, err
+	}
+
+	// fetch biller info - tenant info from tenant id
+	tenant, err := s.TenantRepo.GetByID(ctx, inv.TenantID)
+	if err != nil {
+		return nil, err
+	}
+
+	invoiceData, err := s.getInvoiceData(ctx, inv, customer, tenant)
+	if err != nil {
+		return nil, err
+	}
+
+	// generate pdf
+	return s.PDFGenerator.RenderInvoicePdf(ctx, invoiceData)
+
+}
+
+func (s *invoiceService) getInvoiceData(
+	ctx context.Context,
+	inv *invoice.Invoice,
+	customer *customer.Customer,
+	tenant *tenant.Tenant,
+) (*pdf.InvoiceData, error) {
+	if len(inv.LineItems) == 0 {
+		return nil, ierr.NewError("no line items found").Mark(ierr.ErrSystem)
+	}
```

Comment:
Old invoices in the system with 0$ charges don't have line items so it is a bit of a problem - can add this check later once we have this issue fixed